vllm/v1/engine/async_llm.py

```diff
@@ -220,7 +220,7 @@ async def add_request(
         queue = RequestOutputCollector(output_kind=params.output_kind)
 
         # Convert Input --> Request.
-        prompt_str, request = self.processor.process_inputs(
+        prompt_str, request = await self.processor.process_inputs_async(
```

Comment:
IMHO you could change it to `submit task -- wait done -- transfer data` operation. Then other code will be unchanged 

```python
executor = _get_executor(mm_config.parallel_processor_backend)
prompt_str, request_shm = await asyncio.wrap_future(executor.submit(self.processor.process_inputs, args))
request = restore_from_shm(request_shm)
```